python/ray/tune/integration/pytorch_lightning.py

```diff
@@ -76,16 +78,36 @@ def on_sanity_check_end(self, trainer: Trainer, pl_module: LightningModule):
         if "sanity_check_end" in self._on:
             self._handle(trainer, pl_module)
 
-    def on_epoch_start(self, trainer: Trainer, pl_module: LightningModule):
-        if "epoch_start" in self._on:
+    def on_train_epoch_start(self, trainer: Trainer, pl_module: LightningModule):
+        if "train_epoch_start" in self._on:
             self._handle(trainer, pl_module)
```

Comment:
@richardliaw should we ensure backwards compatible support for PTL < 1.6? 

If so, we can do some versioning checks of `pytorch_lightning` similar to https://github.com/ray-project/ray/pull/27395/files

Reply:
So I looked into implementing this. My proposed solution would be to have an additional class, something like `TuneCallbackBackwardCompat` that is the old class, and then we inherit from either `TuneCallback` or the backward compatible version in the remaining classes `TuneReportCallback`, `_TuneCheckpointCallback`, `TuneReportCheckpointCallback`.

Additionally in the tests, we'd need to test both, right?

Reply:
Yeah we should ensure backwards compat support. Having a backwards compat class seems good to me, thanks!

Reply:
Awesome, I've added those changes in the newest commit